torch/utils/data/sampler.py

```diff
@@ -1,4 +1,6 @@
 import torch
+import random
+import numpy as np
```

Comment:
numpy is not a hard dependency of pytorch, so you shouldn't do this.
2 possibilities:
- either use alternative functions instead of numpy, you can probably use torch operations
- import numpy inside your sampler class.